### PR TITLE
support both : and = as compose envvar separators

### LINKED ISSUE #196: Parsing environment variables with `:`
libcompose could support both separate characters `:` and `=` in envvar because docker config object accepted them. However, kompose doesn't rely on docker but libcompose config object. Thus, kompose can only support `=` at this moment.

Related issue: #173. In the [hygieia](https://raw.githubusercontent.com/capitalone/Hygieia/master/docker-compose.yml) example, there are envvars with `:` and kompose getting panic.

### PR DESCRIPTION
**Pull request by ngtuna**
fix #196 #173

### CODE REVIEW

**Comment by ngtuna** [comment]
could you check it? @surajssd

File: pkg/loader/compose/compose.go [hunk 1]
```diff
@@ -36,14 +36,27 @@ type Compose struct {
...
-	for k, v := range e {
+	var character string
+	for _, e := range envars {
+		//FIXME:(tuna) if envvar string contains both = and :, then = will be first pick up. Consider the case: URL=http://examples.com
+		if strings.Contains(e, "=") {
+			character = "="
+		} else if strings.Contains(e, ":") {
+			character = ":"
+		} else {
+			logrus.Errorf("Invalid environment variable format, only : and = separators are supported")
```
Comment (kadel) [inline pkg/loader/compose/compose.go]: just variable name without `:` or `;` is also valid see: https://docs.docker.com/compose/compose-file/#/environment 
This is something that we should also consider.

File: pkg/loader/compose/compose.go [hunk 2]
```diff
@@ -36,14 +36,27 @@ type Compose struct {
 }
 
 // load environment variables from compose file
-func loadEnvVars(e map[string]string) []kobject.EnvVar {
+func loadEnvVars(envars []string) []kobject.EnvVar {
 	envs := []kobject.EnvVar{}
-	for k, v := range e {
+	var character string
+	for _, e := range envars {
+		//FIXME:(tuna) if envvar string contains both = and :, then = will be first pick up. Consider the case: URL=http://examples.com
```
Comment (ngtuna) [inline pkg/loader/compose/compose.go]: That's correct 👍 . I will update accordingly.

**Comment by ngtuna** [comment]
> But there is one think that this PR should include, and that is test for it.

Yes. I will add follow-up commit

File: pkg/loader/compose/compose.go (hunk 1 above)
Comment (ngtuna) [inline pkg/loader/compose/compose.go]: added https://github.com/kubernetes-incubator/kompose/pull/197/commits/ec3aa3d8afa4a45ea84884c907961f4fadf55c15

**Comment by surajssd** [comment]
@ngtuna I have problem building this branch, can you guys tell me what am I missing?

```bash
$ git remote show tuna
* remote tuna
  Fetch URL: https://github.com/ngtuna/kompose
  Push  URL: https://github.com/ngtuna/kompose
  HEAD branch: master
  Remote branches:
    clean-code        tracked
    envvar_colon      tracked
    global-flag       tracked
[SNIP]
    userguide         tracked
  Local branches configured for 'git pull':
    envvar_colon    merges with remote envvar_colon
    project-context merges with remote project-context
    update-objects  merges with remote update-objects
  Local refs configured for 'git push':
    envvar_colon    pushes to envvar_colon    (up to date)
    master          pushes to master          (fast-forwardable)
    project-context pushes to project-context (up to date)
    update-objects  pushes to update-objects  (up to date)
```

```bash
$ git fetch tuna
```

```bash
$ git checkout envvar_colon 
Switched to branch 'envvar_colon'
Your branch is up-to-date with 'tuna/envvar_colon'.

$ git pull
Already up-to-date.
```

```bash
$ build-kompose 
+ cd /home/hummer/go/src/github.com/kubernetes-incubator/kompose
+ go build -tags experimental -o kompose ./cli/main/
# github.com/kubernetes-incubator/kompose/pkg/loader/bundle
pkg/loader/bundle/bundle.go:95: cannot use p (type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
# github.com/kubernetes-incubator/kompose/pkg/loader/compose
pkg/loader/compose/compose.go:101: cannot use proto (type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
pkg/loader/compose/compose.go:110: cannot use proto (type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
# github.com/kubernetes-incubator/kompose/pkg/transformer/kubernetes
pkg/transformer/kubernetes/k8sutils.go:248: cannot use service (type "github.com/skippbox/kompose/pkg/kobject".ServiceConfig) as type "github.com/kubernetes-incubator/kompose/pkg/kobject".ServiceConfig in argument to transformer.ConfigAnnotations
pkg/transformer/kubernetes/k8sutils.go:274: cannot use service (type "github.com/skippbox/kompose/pkg/kobject".ServiceConfig) as type "github.com/kubernetes-incubator/kompose/pkg/kobject".ServiceConfig in argument to transformer.ConfigAnnotations
pkg/transformer/kubernetes/kubernetes.go:188: cannot use port.Protocol (type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
pkg/transformer/kubernetes/kubernetes.go:207: cannot use port.Protocol (type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
# github.com/skippbox/kompose/pkg/transformer/kubernetes
../../skippbox/kompose/pkg/transformer/kubernetes/k8sutils.go:248: cannot use service (type "github.com/kubernetes-incubator/kompose/pkg/kobject".ServiceConfig) as type "github.com/skippbox/kompose/pkg/kobject".ServiceConfig in argument to transformer.ConfigAnnotations
../../skippbox/kompose/pkg/transformer/kubernetes/k8sutils.go:274: cannot use service (type "github.com/kubernetes-incubator/kompose/pkg/kobject".ServiceConfig) as type "github.com/skippbox/kompose/pkg/kobject".ServiceConfig in argument to transformer.ConfigAnnotations
../../skippbox/kompose/pkg/transformer/kubernetes/kubernetes.go:188: cannot use port.Protocol (type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
../../skippbox/kompose/pkg/transformer/kubernetes/kubernetes.go:207: cannot use port.Protocol (type "github.com/kubernetes-incubator/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol) as type "github.com/skippbox/kompose/vendor/k8s.io/kubernetes/pkg/api".Protocol in field value
```

**Comment by kadel** [comment]
@surajssd this is because @ngtuna's branch is not yet rebased to current master. It still uses old import path `github.com/skippbox/kompose`

**Comment by surajssd** [comment]
oh okay

**Comment by ngtuna** [comment]
Hey @kadel @surajssd I made rebase and added test. Please take a look.

**Comment by surajssd** [comment]
@ngtuna I tried pulling and building it and it LGTM

**Comment by ngtuna** [comment]
merging

**Pull request merged**